### DISCUSSION TITLE
Replace default webui tts with piper tts

### DISCUSSION THREAD

**Discussion by hw-0** [open-webui/open-webui#2113] [General]
**Is your feature request related to a problem? Please describe.**
No.

**Describe the solution you'd like**
Implementation of a high quality tts with low resource usage using [piper tts](https://github.com/rhasspy/piper)

**Describe alternatives you've considered**
I have looked at alternatives such as [parler tts](https://github.com/huggingface/parler-tts) which seems to be higher quality but more complicated and resource intensive

**Additional context**
I think that adding piper tts would be relatively simple to implement and I think would give the webui built in tts a more premium feel

**Comment by pazoff** [comment]
Samples for [Piper](https://github.com/rhasspy/piper), a fast and local text to speech system:
https://rhasspy.github.io/piper-samples/

**Comment by kbuchaniec** [comment]
Piper support would be awesome

**Comment by justinh-rahb** [comment]
Guys. This one has been covered to death already, please use search..
- #126 
- #1224 
- #1491 
- #2168 
- #2256 

Our position is unchanged, Piper can be used via external OpenAI-API spec servers such as [OpenedAI-Speech](https://github.com/matatonic/openedai-speech) if you like. We won't be including it in the base image due to the sporadic nature of its maintainence. The last "release" was in Nov 2023, it had some commits last month, and overall doesn't appear to be under active development anymore.

**Reply by mcbutterpants007** [reply]
I know this post might be a bit old, but I hope it still finds its way to the right people. I've read everything that's been shared, and it's both sad and disheartening to see where Piper has ended up. Looking back, I wonder if I should have listened to my father back in '94 when he suggested I start a career in computers—instead, I chose underwater construction for the US Navy (a story for another time).

Who currently owns Piper? Honestly, I’d love to know. Maybe bringing in some financial incentives could help address these issues. We can’t just acknowledge the problem and leave it at that. In my view, if the tribes don’t come together, we might still be asking a 1970s computer what 2+2 is. We’ve come a long way, and I’d love to see us move the ball closer to the pins.

If you're interested in discussing this further, feel free to reach out.  (FYI AI help me edit this so I don't sound like a 5 year old that cant spell)
